core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -701,7 +701,7 @@ private[spark] object Utils extends Logging {
     }
   }
 
-  private var customHostname: Option[String] = None
+  private var customHostname: Option[String] = sys.env.get("SPARK_LOCAL_HOSTNAME")
```

Comment:
BTW - here is how it is defined now:

```
  <tr>
    <td><code>SPARK_PUBLIC_DNS</code></td>
    <td>Hostname your Spark program will advertise to other machines.</td>
  </tr>
```

In practice though it looks like it might only be used for the UI links.